packages/agreement-platformstate-writer/src/utils.ts

```diff
@@ -93,3 +111,431 @@ export const deleteAgreementEntry = async (
+  const runPaginatedQuery = async (
+    consumerId_eserviceId: GSIPKConsumerIdEServiceId,
+    dynamoDBClient: DynamoDBClient,
+    exclusiveStartKey?: Record<string, AttributeValue>
+  ): Promise<TokenGenerationStatesClientPurposeEntry[]> => {
+    const input: QueryInput = {
+      TableName: config.tokenGenerationReadModelTableNameTokenGeneration,
+      IndexName: "Agreement",
+      KeyConditionExpression: `GSIPK_consumerId_eserviceId = :gsiValue`,
+      ExpressionAttributeValues: {
+        ":gsiValue": { S: consumerId_eserviceId },
+      },
+      ExclusiveStartKey: exclusiveStartKey,
+    };
+    const command = new QueryCommand(input);
+    const data: QueryCommandOutput = await dynamoDBClient.send(command);
+
+    if (!data.Items) {
+      throw genericInternalError(
+        `Unable to read token state entries: result ${JSON.stringify(data)} `
+      );
+    } else {
+      const unmarshalledItems = data.Items.map((item) => unmarshall(item));
+
+      const tokenStateEntries = z
+        .array(TokenGenerationStatesClientPurposeEntry)
+        .safeParse(unmarshalledItems);
+
+      if (!tokenStateEntries.success) {
+        throw genericInternalError(
+          `Unable to parse token state entry item: result ${JSON.stringify(
+            tokenStateEntries
+          )} - data ${JSON.stringify(data)} `
+        );
+      }
+
+      await updateAgreementStateOnTokenStatesEntries({
+        entriesToUpdate: tokenStateEntries.data,
+        agreementState,
+        dynamoDBClient,
+      });
+
+      if (!data.LastEvaluatedKey) {
+        return tokenStateEntries.data;
+      } else {
+        return [
+          ...tokenStateEntries.data,
+          ...(await runPaginatedQuery(
+            consumerId_eserviceId,
+            dynamoDBClient,
+            data.LastEvaluatedKey
+          )),
+        ];
+      }
+    }
+  };
+
+  return await runPaginatedQuery(
+    GSIPK_consumerId_eserviceId,
+    dynamoDBClient,
+    undefined
+  );
+};
+
+export const readCatalogEntry = async (
+  primaryKey: PlatformStatesEServiceDescriptorPK,
+  dynamoDBClient: DynamoDBClient
+): Promise<PlatformStatesCatalogEntry | undefined> => {
+  const input: GetItemInput = {
+    Key: {
+      PK: { S: primaryKey },
+    },
+    TableName: config.tokenGenerationReadModelTableNamePlatform,
+  };
+  const command = new GetItemCommand(input);
+  const data: GetItemCommandOutput = await dynamoDBClient.send(command);
+
+  if (!data.Item) {
+    return undefined;
+  } else {
+    const unmarshalled = unmarshall(data.Item);
+    const catalogEntry = PlatformStatesCatalogEntry.safeParse(unmarshalled);
+
+    if (!catalogEntry.success) {
+      throw genericInternalError(
+        `Unable to parse catalog entry item: result ${JSON.stringify(
+          catalogEntry
+        )} - data ${JSON.stringify(data)} `
+      );
+    }
+    return catalogEntry.data;
+  }
+};
+
+export const isAgreementTheLatest = async (
```

Comment:
Instead of importing it as `isLatestAgreement` why don't we change its name here? I think that `isLatestAgreement` sounds better

Reply:
Sorry, I didn't realize that the renaming operation did that. Of course, I meant to rename the actual function. Fixed